lib/types/gpt.nix

```diff
@@ -55,7 +55,13 @@ in
           };
           label = lib.mkOption {
             type = lib.types.str;
-            default = "${config._parent.type}-${config._parent.name}-${partition.config.name}";
+            default = let
+              limit = 36; # not sure why it's the limit but it seems to break with 37 characters
```

Comment:
For GPT it's actually 72 characters inside the label?

Reply:
Ah, found it: https://en.wikipedia.org/wiki/GUID_Partition_Table


56 (0x38) 	72 bytes 	Partition name (36 [UTF-16](https://en.wikipedia.org/wiki/UTF-16)LE code units)

Reply:
So fdisk encodes the name as UTF-16.